rules/subrepo_rules.build_defs

```diff
@@ -270,3 +270,41 @@ def local_repository(name:str, path:str):
         name = name,
         path = path,
     )
+
+def plugin_repo(name:str, revision:str,  plugin:str="", owner:str="please-build", hashes:list=[]):
+    """Defines a please plugin to use from the configured plugin repositories. See
+        https://please.build/config.html#please.pluginrepositories for more information.
+
+    Args:
+        name (str): Name of the rule
+        revision (str): The revision or version of the plugin pass to the template URLS.
+        plugin (str): The name of the plugin. Defaults to the name of this rule.
+        owner (str): The owner of the plugin to be passed to the template URLS.
+        hashes (list): A list of valid hashes for the downloaded archive file.
+    """
+    plugin = plugin or name
+    urls = [url.format(owner=owner, revision=revision, plugin=plugin) for url in CONFIG.PLUGIN_REPOS]
+
+    remote_rule = remote_file(
+        name = name,
+        _tag = 'download',
+        url = urls,
+        out = name + '_' + basename(urls[0]),
+        hashes = hashes,
+    )
+    extract_rule = build_rule(
+        name = name,
+        srcs = [remote_rule],
+        tools = [CONFIG.JARCAT_TOOL],
+        outs = [name],
+        cmd = " && ".join([
+          "$TOOL x $SRCS -o _tmp",
+          "mv $(dirname $(find . -name .plzconfig | sort | head -n 1)) $OUT",
```

Comment:
This provides some niceties so that the download just has to contain a please repo somewhere inside it.